java/pattern/schema/Rule.java

```diff
@@ -163,8 +164,14 @@ private static void validateThen(String label, @Nullable Conjunction<? extends P
                 .reduce(true, Boolean::logicalAnd)) {
             throw TypeQLException.of(INVALID_RULE_THEN_ROLES.message(label, then));
         }
+
+        // relation variables used to infer new relations in then clause should be anonymous
+        if (then.relation().isPresent()&&then.reference().isName()) {
+                throw TypeQLException.of(RELATION_IN_THEN_NOT_ANONYMOUS.message(label, then));
```

Comment:
Careful. You've copied the call from a message that has two format specifiers ('%s'), but yours doesn't have any.
(Also, please apply auto-spacing with Cmd+L)